dandi/metadata.py

```diff
@@ -467,35 +486,40 @@ def extract_species(metadata):
             )
         return models.SpeciesType(identifier=value_id, name=value)
     else:
-        return ...
+        return None
 
 
-def extract_assay_type(metadata):
+def extract_assay_type(metadata: dict) -> Optional[List[models.AssayType]]:
     if "assayType" in metadata:
         return [models.AssayType(identifier="assayType", name=metadata["assayType"])]
     else:
-        return ...
+        return None
 
 
-def extract_anatomy(metadata):
+def extract_anatomy(metadata: dict) -> Optional[List[models.Anatomy]]:
     if "anatomy" in metadata:
         return [models.Anatomy(identifier="anatomy", name=metadata["anatomy"])]
     else:
-        return ...
+        return None
+
+
+M = TypeVar("M", bound=models.DandiBaseModel)
 
 
-def extract_model(modelcls, metadata, **kwargs):
-    m = modelcls.unvalidated()
+def extract_model(modelcls: Type[M], metadata: dict, **kwargs: Any) -> M:
+    m = cast(M, modelcls.unvalidated())
     for field in m.__fields__.keys():
         value = kwargs.get(field, extract_field(field, metadata))
-        if value is not Ellipsis:
+        if value is not None:
```

Comment:
I wonder (didn't do archaeologic expedition) and have some vague memory that we chose `Ellipsis` for some reason, i.e. that value could really be `None` and thus assigned.  Do you remember/know the reason why Ellipsis was used and why it could be RFed safely now to become good simple `None`?

Reply:
As far as I recall, `Ellipsis` was used because we wanted extracted fields that evaluated to `None` to still be assigned `None`.  However, this was apparently unnecessary, as all of the fields in question default to `None` anyway when unset, so not assigning extracted `None`s doesn't create any problems.

Also, for the record, the reason I had to drop the use of `Ellipsis` here was because it can't be used as a type in annotations prior to Python 3.10.